src/main/java/io/odpf/firehose/sink/bigquery/BigQuerySinkFactory.java

```diff
@@ -0,0 +1,83 @@
+package io.odpf.firehose.sink.bigquery;
+
+import com.gojek.de.stencil.StencilClientFactory;
+import com.gojek.de.stencil.client.StencilClient;
+import com.gojek.de.stencil.parser.Parser;
+import com.gojek.de.stencil.parser.ProtoParser;
+import com.gojek.de.stencil.parser.ProtoParserWithRefresh;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.TransportOptions;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.TableId;
+import io.odpf.firehose.config.BigQuerySinkConfig;
+import io.odpf.firehose.config.enums.SinkType;
+import io.odpf.firehose.metrics.Instrumentation;
+import io.odpf.firehose.metrics.StatsDReporter;
+import io.odpf.firehose.sink.Sink;
+import io.odpf.firehose.sink.SinkFactory;
+import io.odpf.firehose.sink.bigquery.converter.MessageRecordConverterCache;
+import io.odpf.firehose.sink.bigquery.handler.BigQueryClient;
+import io.odpf.firehose.sink.bigquery.handler.BigQueryRow;
+import io.odpf.firehose.sink.bigquery.handler.BigQueryRowWithInsertId;
+import io.odpf.firehose.sink.bigquery.handler.BigQueryRowWithoutInsertId;
+import io.odpf.firehose.sink.bigquery.proto.ProtoUpdateListener;
+import org.aeonbits.owner.ConfigFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Map;
+
+public class BigQuerySinkFactory implements SinkFactory {
+
+    @Override
+    public Sink create(Map<String, String> env, StatsDReporter statsDReporter, StencilClient defaultClient) {
+        BigQuerySinkConfig sinkConfig = ConfigFactory.create(BigQuerySinkConfig.class, env);
+        try {
+            BigQuery bq = getBigQueryInstance(sinkConfig);
+            BigQueryClient bigQueryClient = new BigQueryClient(bq, sinkConfig);
+            MessageRecordConverterCache recordConverterWrapper = new MessageRecordConverterCache();
+            ProtoUpdateListener protoUpdateListener = new ProtoUpdateListener(sinkConfig, bigQueryClient, recordConverterWrapper);
+            StencilClient client = sinkConfig.isSchemaRegistryStencilEnable()
+                    ? (sinkConfig.isAutoSchemaUpdateEnabled()
+                               ? StencilClientFactory.getClient(sinkConfig.getSchemaRegistryStencilUrls(), env, statsDReporter.getClient(), protoUpdateListener)
+                               : StencilClientFactory.getClient(sinkConfig.getSchemaRegistryStencilUrls(), env, statsDReporter.getClient()))
+                    :
+                    StencilClientFactory.getClient();
+
+            Parser parser = sinkConfig.getAutoRefreshCache()
+                    ? new ProtoParser(client, sinkConfig.getInputSchemaProtoClass())
+                    : new ProtoParserWithRefresh(client, sinkConfig.getInputSchemaProtoClass());
```

Comment:
ProtoParserWithRefresh will go away if we remove that config.

Reply:
yes. removing this